agir/events/views/api_views.py

```diff
@@ -231,11 +232,14 @@ def post(self, request, pk):
         return JsonResponse({"data": True})
 
 
-class CancelEventAPIView(APIView):
+class CancelEventAPIView(GenericAPIView):
     permission_classes = (EventManagementPermissions,)
-    queryset = Event.objects.all()
+    queryset = Event.objects.upcoming(as_of=timezone.now(), published_only=False)
 
     def post(self, request, pk):
+        self.object = self.get_object()
+        self.check_object_permissions(request, self.object)
+
```

Comment:
```suggestion
        self.get_object()
```
Je crois que `get_object` appelle aussi `check_object_permissions`, normalement il n'y a pas besoin de le rappeler (et il n'y pas besoin de l'enregistrer dans `self.object` non plus)